src/NBitcoin/Payment/PaymentACK.cs

```diff
@@ -79,26 +71,52 @@ public void WriteTo(Stream output)
         {
             Protobuf.ProtobufReaderWriter proto = new ProtobufReaderWriter(output);
             proto.WriteKey(1, ProtobufReaderWriter.PROTOBUF_LENDELIM);
-            proto.WriteBytes(Payment.ToBytes());
-            if(Memo != null)
+            proto.WriteBytes(this.Payment.ToBytes());
+            if (this.Memo != null)
             {
                 proto.WriteKey(2, ProtobufReaderWriter.PROTOBUF_LENDELIM);
-                proto.WriteString(Memo);
+                proto.WriteString(this.Memo);
             }
         }
 #if !NOFILEIO
         public static PaymentACK Load(string file)
         {
-            using(var fs = File.OpenRead(file))
+            using (FileStream fs = File.OpenRead(file))
             {
                 return Load(fs);
             }
         }
 #endif
     }
+
     public class PaymentMessage
     {
         public const int MaxLength = 50000;
+        public readonly static string MediaType = "application/bitcoin-payment";
+
+        private readonly List<PaymentOutput> refundTo = new List<PaymentOutput>();
+        private readonly List<Transaction> transactions = new List<Transaction>();
+
+        public string Memo { get; set; }
+        public byte[] MerchantData { get; set; }
+        public Uri ImplicitPaymentUrl { get; set; }
+
+        public List<Transaction> Transactions
```

Comment:
Ah can't we just single this?

`public List<Transaction> Transactions { get { return this.transactions; } }`

Reply:
Changed.